src/views/product-landing/server.ts

```diff
@@ -1,108 +1,86 @@
-        }
-        usedHeadings.push(slug)
-        return { ...block, slug }
-      case 'cards':
-        // TODO: instead of the below, should likely use product context
-        // TODO: in IconTile for default brandColor (rather than current
-        // TODO: behavior of static default to "neutral"). Would eliminate
-        // TODO: the need to add iconBrandColor to cards (unless doing so
-        // TODO: as an explicit override).
-        // ensure cards with icons have an iconBrandColor,
-        // falling back to the current product if not set
-        /* eslint-disable-next-line no-case-declarations */
-        let defaultIconColor
-        if (product.slug === 'hcp' || product.slug === 'sentinel') {
-          defaultIconColor = 'neutral'
-        } else {
-          defaultIconColor = product.slug
-        }
-        return {
-          ...block,
-          cards: block.cards.map((card) => ({
-            ...card,
-            iconBrandColor: card.iconBrandColor || defaultIconColor,
-          })),
-        }
-      default:
-        return block
-    }
-  })
+  /**
+   * Validate that CONTENT matches our schema. This includes a type guard,
+   * which asserts that CONTENT is ProductLandingContent.
+   */
+  validateAgainstSchema<ProductLandingContent>(
+    CONTENT,
+    ProductLandingContentSchema,
+    jsonFilePath
+  )
 
   /**
-   * Constructs the levels of nav data used in the `Sidebar` on all
-   * `ProductLandingView` pages.
+   * Transform content to props.
+   * This includes filling in inline tutorials and collection content.
    */
-  const sidebarNavDataLevels = [
-    generateTopLevelSidebarNavData(product.name),
-    generateProductLandingSidebarNavData(product),
-  ]
+  const content = await transformRawContentToProp(CONTENT, product)
 
+  /**
+   * Gather up our static props package & return it
+   */
   return {
-    content: CONTENT,
+    content,
+    product,
     layoutProps: {
-      headings: CONTENT.blocks
-        .filter((s) => s.type == 'heading')
-        .map(({ heading, slug, level }) => ({
-          title: heading,
-          slug: slug,
-          level,
-        })),
+      headings: extractHeadings(content),
       breadcrumbLinks: [
         { title: 'Developer', url: '/' },
         { title: product.name, url: `/${product.slug}` },
       ],
-      sidebarNavDataLevels,
+      /**
+       * @TODO remove casting to `any` (used $TSFixMe here for visibility).
+       * This requires refactoring both `generateTopLevelSidebarNavData` and
+       * `generateProductLandingSidebarNavData` to set up `menuItems` with the
+       * correct types.
+       * This is outside the scope of the product landing page content build,
+       * so deferring to a sidebar-focused follow-up PR.
+       */
+      sidebarNavDataLevels: [
+        generateTopLevelSidebarNavData(product.name),
+        generateProductLandingSidebarNavData(product),
+      ] as $TSFixMe,
```

Comment:
Update relative to assembled PRs: the return type for this `server.ts` function was previously `$TSFixMe`, so we didn't run into issues with typing for `sidebarNavDataLevels`. With the more explicit return type after the product-landing refactor, we do run into type issues.

I think resolving these issues requires underlying work in sidebar (as mentioned in similar comments added in #408), so I think this type work falls outside the scope of this PR.